autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -190,14 +205,22 @@ public Set<Long> findAllInStatusesThatStartedBefore(Collection<ActivityStatus> s
     public void deleteScalingActivityByIds(Set<Long> activityIds) {
         LOGGER.info("Deleting {} scalingActivities by their Ids", activityIds.size());
         scalingActivityRepository.deleteAllById(activityIds);
+        calculateUnprocessedScalingActivityMetrics();
     }
 
     public void deleteScalingActivityForCluster(Long clusterId) {
         LOGGER.info("Deleting all scaling activity for clusterId: {}", clusterId);
         scalingActivityRepository.deleteAllByCluster(clusterId);
     }
 
-    public void save(ScalingActivity scalingActivity) {
-        scalingActivityRepository.save(scalingActivity);
+    public ScalingActivity save(ScalingActivity scalingActivity) {
+        return scalingActivityRepository.save(scalingActivity);
+    }
+
+    private void calculateUnprocessedScalingActivityMetrics() {
+        // TODO: Should we fetch a broader collection of scaling events?
```

Comment:
This, I think, is the opposite of what we want to track. The events which are applicable for cleanup - will be cleaned up (assuming the system is not too busy).

What would be useful to track is: 
1) How many operations exist which are waiting for an activity to complete (i.e. are not cleanup candidates) - and effectively cause the table to grow continuously.
2) The total size of this table
3) Maybe this metric: the total number of cleanup candidates.

Reply:
Tracked all these metrics